prefect_kubernetes/jobs.py

```diff
@@ -299,3 +309,205 @@ def kubernetes_orchestrator():
             namespace=namespace,
             **kube_kwargs,
         )
+
+
+class KubernetesJobRun(JobRun[Dict[str, Any]]):
+    """A container representing a run of a Kubernetes job."""
+
+    def __init__(self, kubernetes_job: "KubernetesJob"):
+        self._kubernetes_job = kubernetes_job
+        self.pod_logs = {}
+
+    async def wait_for_completion(self):
+        """Waits for the job to complete.
+
+        Raises:
+            RuntimeError: If the Kubernetes job fails.
+        """
+        with self._kubernetes_job.credentials.get_client(
+            "batch"
+        ) as batch_v1_client, self._kubernetes_job.credentials.get_client(
+            "core"
+        ) as core_v1_client:
+            completed = False
+
+            elapsed_time = 0
+
+            while not completed:
+                job_expired = (
+                    elapsed_time > self._kubernetes_job.timeout_seconds
+                    if self._kubernetes_job.timeout_seconds
+                    else False
+                )
+                if job_expired:
+                    raise TimeoutError(f"Job timed out after {elapsed_time} seconds.")
+
+                latest_v1_job = await run_sync_in_worker_thread(
+                    batch_v1_client.read_namespaced_job_status,
+                    name=self._kubernetes_job.v1_job.metadata.name,
+                    namespace=self._kubernetes_job.namespace,
+                    **self._kubernetes_job.api_kwargs,
+                )
+
+                v1_pod_list = await run_sync_in_worker_thread(
+                    core_v1_client.list_namespaced_pod,
+                    namespace=self._kubernetes_job.namespace,
+                    label_selector=(
+                        f"controller-uid="
+                        f"{latest_v1_job.metadata.labels['controller-uid']}"
```

Comment:
This feels like a good candidate to save to a variable since it's two lines and it uses f string to access a value from a dict.